### ISSUE TITLE
ArcGIS Pro version of editor

### ISSUE THREAD

**Issue by mboeringa** [Esri/arcgis-osm-editor#120]
Considering ArcGIS Pro already had its first birthday, it may be time to look at a conversion of the ArcGIS Editor for OpenStreetMap to Pro (besides a [10.4 version](https://github.com/Esri/arcgis-osm-editor/issues/116) of the toolbox).

If a full conversion to Pro is to much of an effort at this point in time, especially considering the advanced editing and posting functionality regarding the OSM main database, it might be an idea to develop a lightweight OSM conversion & dataloader instead as a first step to a full conversion of the Editor.

Having a basic OSM data conversion tool in Pro would already be of much use to many people starting to explore Pro, and would facilitate my ArcGIS Renderer for OpenStreetMap as well.

Such a lightweight "ArcGIS Data Loader for OpenStreetMap" might be based on a subset of the existing tools, most likely:

- Download OSM Data (XAPI)
- Load OSM File (based on the new parallel loader as per https://github.com/Esri/arcgis-osm-editor/pull/110 and https://github.com/Esri/arcgis-osm-editor/pull/119)
- OSM Attribute Selector

and possibly some of the basic symbolization options:

- Symbolize OSM Data
- Symbolize Points
- Symbolize Lines
- Symbolize Polygons

which can probably be used "as is" in Pro once the lyr files have been replaced by lyrx.

After this, a full conversion of the Editor toolbox, could be done on a later date when convenient and time allows.

**Comment by esong27** [comment]
Have you seen any progress on this since? I am trying to convert osm data to Arcgis Pro but cannot find any way to do so...

**Comment by mboeringa** [comment]
> Have you seen any progress on this since? I am trying to convert osm data to Arcgis Pro but cannot find any way to do so...

No, I haven't had any formal response to this one yet as you can see in this thread, although I guess this is on ESRI's TO-DO list or at the very least in the back of their minds while working on other minor updates to the toolbox like the new parallel loader. However, considering the Editor is not an officially supported product, it may take some time as not being part of any official release policies of ESRI.

However, regarding your question: if you have Pro, you also should be able to install ArcMap. That will allow you to do the conversion with the current toolbox, and then use Pro's option for importing ArcMap documents to get the data in Pro (if you intend to stylize in ArcMap first), or simply add the Feature Classes straight to Pro after conversion with ArcMap.

HOWEVER(!): In order for that to function, and depending on what version of ArcMap and the Editor you are running, you _may_ need to have to use the **Remove OSM Editor Extension** tool first on the data you loaded in your geodatabase, to get Pro capable of reading the data. This is not the case with the newer versions of the toolbox afaik. Anyway, if you have a problem with loading the converted data in Pro, try running the tool.

With that said, you really shouldn't have problems getting the data read and viewed in Pro, I have done that multiple times without issues.

**Comment by cloveman** [comment]
We’ve heard the request for the tools to function in ArcGIS Pro, however we’ve not taken any significate actions toward that as of yet. As noted in this thread, you can use the existing tools in ArcMap and Import the mxd after the OSM data has been copied down and symbolized.

**Comment by mboeringa** [comment]
> We’ve heard the request for the tools to function in ArcGIS Pro, however we’ve not taken any significate actions toward that as of yet. As noted in this thread, you can use the existing tools in ArcMap and Import the mxd after the OSM data has been copied down and symbolized.

Hi @cloveman ,

Thanks for joining in on this one.

To be honest, your answer still leaves a lot of questions open. I realize a full Editor conversion, considering the low usage stats and excellent alternative editors like iD, JOSM etc, may be a low priority for ESRI.

I would therefore like you to focus and comment regarding the secondary possibility I have been posing here: that is, no Pro version of the full Editor, but just the import tools, and specifically the new fast parallel loader @ThomasEmge has been doing such excellent work on - the **OSM File Loader (load only)** tool - combined with the **Download OSM Data (XAPI)** and **OSM Attribute Selector** tool.

Although I am unfamiliar with the exact challenges such a toolbox conversion from ArcMap to Pro may incur, I think it likely to be far less demanding, and still provide the majority of ArcGIS Pro users interested in OSM data, with a great small toolset allowing easy access to the data and a great conversion option. I think the combination of these three core tools will provide great value to many of them.

**Comment by cloveman** [comment]
Yes, that is helpful and potentially a good way for us to get started with a subset of the tools, and I agree with the greater value of the specific tools you identified.

**Comment by Renlong-Yin** [comment]
After one year the editor still have NOT been realised, what could I deal it with, ESRI, when I need to download and symbolize OSM file??!

**Comment by Chaz6** [comment]
Can we get an update on this please? I have been updated to ArcGIS Pro 2.0 and I no longer have access to ArcGIS Desktop, which leaves me currently unable to generate a network dataset from OSM.

**Comment by cloveman** [comment]
@Chaz6 we are in the process of evaluating the LOE to do this, there is no timeline as of yet.

**Comment by yobimania** [comment]
@cloveman any progress on a ArcGIS Pro editor since Feb 18?

**Comment by skmoore** [comment]
@yobimania We don't have any plans to migrate this toolset to Pro, however we are looking at ways we could make OSM data accessible in Pro.

We've found that most users of this toolset don't actually use it to edit features into OSM. Are you just looking to get OSM data into a file geodatabase?

**Comment by yobimania** [comment]
@skmoore pretty much just need it in gdb format however, as a contributor to OSM I find the web interface to be quite restrictive in making large amounts edits quickly when aspatial data manipulation in ArcGIS can be done much faster (an example being geocoding residential polygons) which currently is an extremely monotonous task of manual data entry.

**Comment by skmoore** [comment]
@yobimania Ok, sounds like an extraction tool in Pro would mostly suit your needs then.

For the editing issues you have, you might want to try using [JOSM](https://josm.openstreetmap.de/). I'm not very familiar with it but I know it's much better suited than iD for large edits like building footprints.

We also have a pending [pull request on the iD editor](https://github.com/openstreetmap/iD/pull/4633) repository to allow geometries to be imported from REST services. So if you had access to a dataset like residential polygons which was already published as a service, you could automatically import the geometries into the editor rather than having to manually digitize them one by one. There's been some back and forth discussion on it, we're waiting to see what happens for now.

**Comment by d-wasserman** [comment]
I would like to also ask for a network data set template based on OSM networks. We have had a huge amount of success with our pre-processed OSM networks  for network data-set creation and analysis. As far as I know, this really just involves attribution of costs/restrictions and planarizing the networks anywhere tunnels/bridges are not yes and the highway tag is not null (for ways vs. areas).

I would like to see progress on this front as well. At least with a basic tag schema for roads.

**Comment by Chaz6** [comment]
> We've found that most users of this toolset don't actually use it to edit features into OSM. Are you just looking to get OSM data into a file geodatabase?

I need to build network datasets for solving with one-way, barriers, etc.

**Comment by anmolter** [comment]
Any news on this? I also need to build network datasets based on OSM data.

**Comment by TheCre8r** [comment]
With the lifecycle of 10.x, I also would like to request this to be converted.

**Comment by AF02** [comment]
In 2022, still nothing on this application compatible with arcgis pro?
